BorrowBookUI.java

```diff
@@ -2,116 +2,116 @@
 
 
 public class BorrowBookUI {
-	
-	public static enum UI_STATE { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
-
-	private BorrowBookControl CONTROL;
-	private Scanner input;
-	private UI_STATE StaTe;
-
-	
-	public BorrowBookUI(BorrowBookControl control) {
-		this.CONTROL = control;
-		input = new Scanner(System.in);
-		StaTe = UI_STATE.INITIALISED;
-		control.setUI(this);
-	}
-
-	
-	private String input(String prompt) {
-		System.out.print(prompt);
-		return input.nextLine();
-	}	
-		
-		
-	private void output(Object object) {
-		System.out.println(object);
-	}
-	
-			
-	public void Set_State(UI_STATE STATE) {
-		this.StaTe = STATE;
-	}
-
-	
-	public void run() {
-		output("Borrow Book Use Case UI\n");
-		
-		while (true) {
-			
-			switch (StaTe) {			
-			
-			case CANCELLED:
-				output("Borrowing Cancelled");
-				return;
-
-				
-			case READY:
-				String MEM_STR = input("Swipe member card (press <enter> to cancel): ");
-				if (MEM_STR.length() == 0) {
-					CONTROL.cancel();
-					break;
-				}
-				try {
-					int Member_ID = Integer.valueOf(MEM_STR).intValue();
-					CONTROL.Swiped(Member_ID);
-				}
-				catch (NumberFormatException e) {
-					output("Invalid Member Id");
-				}
-				break;
-
-				
-			case RESTRICTED:
-				input("Press <any key> to cancel");
-				CONTROL.cancel();
-				break;
-			
-				
-			case SCANNING:
-				String Book_Str = input("Scan Book (<enter> completes): ");
-				if (Book_Str.length() == 0) {
-					CONTROL.Complete();
-					break;
-				}
-				try {
-					int BiD = Integer.valueOf(Book_Str).intValue();
-					CONTROL.Scanned(BiD);
-					
-				} catch (NumberFormatException e) {
-					output("Invalid Book Id");
-				} 
-				break;
-					
-				
-			case FINALISING:
-				String Ans = input("Commit loans? (Y/N): ");
-				if (Ans.toUpperCase().equals("N")) {
-					CONTROL.cancel();
-					
-				} else {
-					CONTROL.Commit_LOans();
-					input("Press <any key> to complete ");
-				}
-				break;
-				
-				
-			case COMPLETED:
-				output("Borrowing Completed");
-				return;
-	
-				
-			default:
-				output("Unhandled state");
-				throw new RuntimeException("BorrowBookUI : unhandled state :" + StaTe);			
-			}
-		}		
-	}
-
-
-	public void Display(Object object) {
-		output(object);		
-	}
+
+    public static enum UIState { INITIALISED, READY, RESTRICTED, SCANNING, IDENTIFIED, FINALISING, COMPLETED, CANCELLED };
+
+    private BorrowBookControl control;
+    private Scanner input;

```

Comment:
D3: use a name more meaningful than 'input'
(HINT: 'systemInInput' is fine, it tells the reader what the Scanner is scanning)

Reply:
updated Scanner to provided HINT